docs/API-functions.md

```diff
@@ -459,6 +502,126 @@ setting the `SRTO_RENDEZVOUS` option to true, and doing `srt_connect`.
+groups. No such group type is currently implemented.
+
+### srt_exclude
+
+```
+int srt_exclude(SRTSOCKET socket);
+```
+This function removes a socket from a group to which it currently belongs.
+This is only allowed for unmanaged groups. No such group type is currently
+implemented.
+
+### srt_groupof
+
+```
+SRTSOCKET srt_groupof(SRTSOCKET socket);
+```
+
+Returns the group ID of the socket, or `SRT_INVALID_SOCK` if the socket
+doesn't exist or it's not a member of any group.
+
+### srt_group_data 
+
+```
+int srt_group_data(SRTSOCKET socketgroup, SRT_SOCKGROUPDATA* output, size_t* inoutlen);
+```
+
+* `socketgroup` an existing socket group ID
+* `output` points to an output array
+* `inoutlen` points to a variable set to array's size
+
+This function obtains the current member state of the group specified in
+`socketgroup`. The `output` should point to the array of enough size to get
+all the elements and `inoutlen` to a variable preset to the size of this array.
+The current number of members will be written back to `inoutlen`. If the size
+is enough for the current number of members, the `output` array will be
+filled with group data and the function will return 0. Otherwise the array
+will not be filled and `SRT_ERROR` will be returned.
+
+- Returns:
+
+   * 0, if successful
+   * -1, on error
+
+- Errors:
+
+   * `SRT_EINVPARAM` reported if `socketgroup` is not an existing group ID
+   * `SRT_SUCCESS` if the array was too small
+
+
+
+### srt_connect_group
+
+```
+int srt_connect_group(SRTSOCKET group,
+                      const struct sockaddr* source /*nullable*/, int sourcelen,
+                      SRT_SOCKGROUPDATA name [], int arraysize);
+```
+
+This function does almost the same as calling `srt_connect` (or
+`srt_connect_bind`, if `source` is not NULL) in a loop for every item specified
+in `name` array. However if you did this in a blocking mode, the first call
+to `srt_connect` would block until the connection is established, whereas this
+function blocks until any of the specified connections is established. 
+
+If you set the group nonblocking mode (`SRTO_RCVSYN` option), there's no
+difference. Note, however, that in this function accepts only groups, not
+sockets.
+
+The elements of the `name` array need to be prepared with the use of the
+`srt_prepare_endpoint` function. Note that it is **NOT** required that every
+address you specify for it is of the same family.
+
+Return value and errors in this function are the same as in `srt_connect`,
+although this function reports success when at least one connection has
+succeeded. Which one and how many of them succeeded, can be checked with the
+`srt_group_data` function.
+
+
+### srt_prepare_endpoint
+
+```
+SRT_API SRT_SOCKGROUPDATA srt_prepare_endpoint(const struct sockaddr* adr, int namelen);
+```
+
+This function turns the given address into `SRT_SOCKGROUPDATA` structure needed by
+`srt_connect_group` function.
```

Comment:
This function turns the given address into the `SRT_SOCKGROUPDATA` structure needed 
by the `srt_connect_group` function.